Add byte array to numeric new functions + UINT256 support (#232)
duneanalytics/docs: docs/reference/dune-v2/query-engine.md

| <div style="width:290px">**Description**</div> | **V1 - PostgreSQL** | **V2 - Spark SQL** | **V2 - Dune SQL** |
| --- | --- | --- | --- |

> | **`bytea2numeric` does not exist in Spark.** | `bytea2numeric` (bytea) | `bytea2numeric_v2` (string) | `bytearray_to_integer` (hex) <br> `bytearray_to_bigint` (hex) <br> `bytearray_to_decimal` (hex) <br> `bytearray_to_uint256` (hex) <br> `bytea2numeric` (hex) <br> More details on [Byte Array to Numeric Functions](#byte-array-to-numeric-functions)|

Reviewer: Add deprecation notice on bytearray2numeric

Reviewer: Or omit it

For example, referencing a column name in the `WHERE` clause using double quotes works as expected. However, the same query inside a CTE treats the column name as a string, [as can be seen here](https://dune.com/queries/1199604).

## Numerical types

> We support the [numerical types](https://trino.io/docs/current/language/types.html) `INTEGER`, `BIGINT`, `DOUBLE`, and fixed precision `DECIMAL` with precision up to 38 digits (i..e, `DECIMAL(38, 0)`). Additionally, we have `UINT256` for doing arithmetic on unsigned 256 bit integers.

Reviewer: ```suggestion
We support the [numerical types](https://trino.io/docs/current/language/types.html) `INTEGER`, `BIGINT`, `DOUBLE`, and fixed precision `DECIMAL` with precision up to 38 digits (i..e, `DECIMAL(38, 0)`). Additionally, we support `UINT256` for representing unsigned 256 bit integers.
```

## Byte Array to Numeric Functions
- `bytearray_to_integer`, returns the `INTEGER` value of a big-endian byte array of length <= 4 representing the integer in two's complement. If the byte array has length < 4 it is padded with zero bytes.
- `bytearray_to_bigint`, returns the `BIGINT` value of a big-endian byte array of length <= 8 representing the bigint in two's complement. If the byte array has length < 8 it is padded with zero bytes.
- `bytearray_to_decimal`, returns the `DECIMAL(38,0)` value of a big-endian byte array of length <= 16 representing the decimal(38,0) in two's complement. If the byte array has length < 16 it is padded with zero bytes.
- `bytearray_to_uint256`, returns the `UINT256` of a big-endian byte array of length <= 32 representing the unsigned integer. If the byte array has length < 32 it is padded with zero bytes.

> - `bytea2numeric` behaves exactly as `bytearrat_to_bigint`

Reviewer: Add deprecation notice. Rewrite:
`bytea2numeric` is deprecated. It is an alias for `bytearray_to_bigint`.

> The functions throw an overflow exception if the byte array is larger than the number of bytes supported for the type, even if the most significant bytes are all zero. For this cases, it is possible to use `bytearray_ltrim` in order to trim the left zero bytes.

Reviewer: ```suggestion
The functions throw an overflow exception if the byte array is larger than the number of bytes of the type, even if the most significant bytes are all zero. It is possible to use `bytearray_ltrim` to trim the zero bytes from the left.
```

> The functions throw an overflow exception if the byte array is larger than the number of bytes supported for the type, even if the most significant bytes are all zero. For this cases, it is possible to use `bytearray_ltrim` in order to trim the left zero bytes.

Reviewer: ```suggestion
The bytearray conversion functions throw an overflow exception if the byte array is larger than the number of bytes supported for the type, even if the most significant bytes are all zero. For this cases, it is possible to use `bytearray_ltrim` in order to trim the left zero bytes.
```
